### PR TITLE
Edit formations

### PR DESCRIPTION
**Pull request by KMontag42**
allows editing formations
loading screen for every route segment, makes the app feel much snappier
`npm update`
`trunk update`

### CODE REVIEW

File: app/formations/[id]/edit/page.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  }
+
+  const { userId } = auth();
+
+  if (!userId) {
+    auth().redirectToSignIn();
+    return;
+  }
+
+  if (formation.user_id !== userId) {
```
Comment (SwiftieHusband) [inline app/formations/[id]/edit/page.tsx:30]: Do you need this? Shouldn't we put this stuff in middleware to catch for specific pages if they aren't logged in? The only case here would be if the `userId` that you're logged in as isn't the owner of the formation, then we should redirect from the component?

File: app/formations/[id]/page.tsx [hunk 2]
```diff
@@ -39,9 +41,17 @@ export default async function FormationPage({ params }: Props) {
 
   const cmsData = await getCmsData();
 
+  const isMyFormation = formation.user_id === userId;
+
   return (
     <div className="container md:w-[40vw]">
-      <FormationCard data={formation as any} cmsData={cmsData} />
+      <FormationCard
+        data={formation as any}
```
Comment (SwiftieHusband) [inline app/formations/[id]/page.tsx:49]: This could be specified as `FormationData` instead of `any`

File: app/formations/mine/page.tsx [hunk 3]
```diff
@@ -25,8 +25,10 @@ export default async function MyFormations() {
 
   return (
     <div className="container">
-      <Link href={`/users/${username}`} className="flex justify-center pb-4">
-        <Button>View your profile</Button>
+      <Link href={`/users/${username}`}>
+        <Button variant={"analytica"} className="mb-4">
```
Comment (SwiftieHusband) [inline app/formations/mine/page.tsx:29]: technically you don't need `{}` here.. nit picky, I don't like seeing them unless you need them

File: app/formations/[id]/edit/page.tsx (hunk 1 above)
Comment (KMontag42) [reply app/formations/[id]/edit/page.tsx:30]: i couldn't find good documentation on the clerk side about how to catch dynamic route segments in their middleware.

this particular check is for when someone doesn't own the component.
we can redirect away from the page for sure.

File: app/formations/mine/page.tsx (hunk 3 above)
Comment (KMontag42) [reply app/formations/mine/page.tsx:29]: i only use them b/c that's what my tab complete automatically does xD, not directly needed

File: app/formations/[id]/edit/page.tsx (hunk 1 above)
Comment (SwiftieHusband) [reply app/formations/[id]/edit/page.tsx:30]: I guess my thought is if they hit the `*/edit` route and aren't logged in, they shouldn't even get this far to the component. The middleware would stop this.

Comment (KMontag42) [reply app/formations/[id]/edit/page.tsx:30]: [clerk docs](https://clerk.com/docs/references/nextjs/clerk-middleware#protect-routes)

**Review by SwiftieHusband — APPROVED** [review]

**Pull request merged**